### PR TITLE
Enable auto_mkdir for local filesystem and run on all supported stora…

### PR DESCRIPTION
**Pull request by sultaniman**
This PR adjusts filesystem pipeline tests for local filesystem and enables `auto_mkdir` in tests and runs through all storage backends.

### CODE REVIEW

File: tests/load/pipeline/test_filesystem_pipeline.py [hunk 1]
```diff
@@ -299,30 +301,36 @@ def count(*args, **kwargs) -> Any:
         "hiphip": counter("Hurraaaa"),
     }
     now = pendulum.now()
-    os.environ["DESTINATION__FILESYSTEM__BUCKET_URL"] = "file://_storage"
+    kwargs = {}
+    if default_buckets_env.startswith("file://"):
```
Comment (rudolfix) [inline tests/load/pipeline/test_filesystem_pipeline.py:305]: maybe I'm missing something but we decided to make path just before `put_file` in copy job and only for local filesystem.

**Review by rudolfix — CHANGES_REQUESTED** [review]

File: tests/load/pipeline/test_filesystem_pipeline.py (hunk 1 above)
Comment (sultaniman) [reply tests/load/pipeline/test_filesystem_pipeline.py:305]: @rudolfix sorry I've been busy with other two prs, I will do it soon :)

Comment (sultaniman) [reply tests/load/pipeline/test_filesystem_pipeline.py:305]: @rudolfix I created a separate PR for this https://github.com/dlt-hub/dlt/pull/1267

Comment (sultaniman) [reply tests/load/pipeline/test_filesystem_pipeline.py]: Merged.

File: tests/load/pipeline/test_filesystem_pipeline.py [hunk 2]
```diff
@@ -355,6 +363,38 @@ def count(*args, **kwargs) -> Any:
...
+
+    register_implementation("file", MockLocalFS, clobber=True)
+
+    yield
+
+    register_implementation("file", LocalFileSystem, clobber=True)
+
+
+@pytest.mark.parametrize("layout", TEST_LAYOUTS)
+def test_filesystem_local_filesystem_put_file_fails_if_we_do_not_create_destination_folder_regression(
```
Comment (sultaniman) [inline tests/load/pipeline/test_filesystem_pipeline.py:380]: @rudolfix @sh-rp here is the regression test, in which I replaced local filesystem implementation with my own mocked version and then later checking the exception chain.

**Review by sh-rp — APPROVED** [review]

**Review by rudolfix — APPROVED** [review]
LGTM! and thx for all the fixes

**Pull request merged**